src/main/java/de/srendi/advancedperipherals/common/addons/computercraft/apis/EntityAPI.java

```diff
@@ -0,0 +1,81 @@
+
+import dan200.computercraft.api.lua.*;
+import dan200.computercraft.shared.util.NBTUtil;
+import de.srendi.advancedperipherals.common.configuration.APConfig;
+import de.srendi.advancedperipherals.common.util.EntityUtil;
+import de.srendi.advancedperipherals.common.util.LuaConverter;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.player.Player;
+
+import java.util.Map;
+import java.util.UUID;
+
+public class EntityAPI {
+
+    public static final EntityAPI ENTITY_API = new EntityAPI();
+
+    @LuaFunction(mainThread = true)
+    public final Object getNBT(IArguments arguments) throws LuaException {
+        if (!APConfig.API_CONFIG.enableGetNBT.get())
+            throw new LuaException("This function is disabled in the config. Activate it or ask an admin if they can activate it.");
+        Entity entity = EntityUtil.getEntityFromUUID(UUID.fromString(arguments.getString(0)));
+        if (!APConfig.API_CONFIG.enablePlayerAccess.get() && entity instanceof Player)
+            throw new LuaException("Using players in EntityAPI is disabled in the config. Activate it or ask an admin if they can activate it.");
+        return NBTUtil.toLua(entity.serializeNBT());
+    }
+
+    @LuaFunction(mainThread = true)
+    public final Object getName(IArguments arguments) throws LuaException {
+        Entity entity = EntityUtil.getEntityFromUUID(UUID.fromString(arguments.getString(0)));
+        return entity.getName().getString();
+    }
+
+    @LuaFunction(mainThread = true)
+    public final Map<String, Object> getBoundingBox(IArguments arguments) throws LuaException {
+        if (!APConfig.API_CONFIG.enableGetBoundingBox.get())
+            throw new LuaException("This function is disabled in the config. Activate it or ask an admin if they can activate it.");
+        Entity entity = EntityUtil.getEntityFromUUID(UUID.fromString(arguments.getString(0)));
+        if (!APConfig.API_CONFIG.enablePlayerAccess.get() && entity instanceof Player)
+            throw new LuaException("Using players in EntityAPI is disabled in the config. Activate it or ask an admin if they can activate it.");
+        return LuaConverter.aabbToObject(entity.getBoundingBox());
+    }
+
+    @LuaFunction(mainThread = true)
+    public final Map<String, Object> getPos(IArguments arguments) throws LuaException {
+        if (!APConfig.API_CONFIG.enableGetPos.get())
+            throw new LuaException("This function is disabled in the config. Activate it or ask an admin if they can activate it.");
+        Entity entity = EntityUtil.getEntityFromUUID(UUID.fromString(arguments.getString(0)));
+        if (!APConfig.API_CONFIG.enablePlayerAccess.get() && entity instanceof Player)
+            throw new LuaException("Using players in EntityAPI is disabled in the config. Activate it or ask an admin if they can activate it.");
+        return LuaConverter.vec3ToLua(entity.getPosition(1));
```

Comment:
```suggestion
        return LuaConverter.vec3ToLua(entity.position());
```
I believe you want `position()`, and forge is not good at naming yeah